tests/Integration/Behaviour/Features/Scenario/Order/order_multi_shop.feature

```diff
@@ -288,3 +289,62 @@ Feature: Order from Back Office (BO)
       | total_paid_real          | 6.00   |
       | total_shipping_tax_excl  | 7.0    |
       | total_shipping_tax_incl  | 7.42   |
+
+  Scenario: In All Shop Context, Update product in order
+    # Create Shop Group & Shops
+    When I add a shop group "shopGroup1" with name "Shop Group 1"
+    And I add a shop "shop2" with name "Shop 2" for the group "Shop Group 1"
+    Then I should have 2 shop groups
+    And I should have 1 shop in group "Default"
+    And I should have 1 shop in group "Shop Group 1"
+    # Create Products
+    When shop context "Default" is loaded
+    And there is a product in the catalog named "Product A" with a price of 12.3 and 0 items in stock
+    And product "Product A" cannot be ordered out of stock
+    Then the remaining available stock for product "Product A" should be 0
+    When shop context "Shop 2" is loaded
+    And there is a product in the catalog named "Product A" with a price of 45.6 and 100 items in stock
+    And product "Product A" cannot be ordered out of stock
+    Then the remaining available stock for product "Product A" should be 100
+    # Create Customer
+    # Context : Shop 2
+    When there is a customer named "testCustomerShop2" whose email is "pubshop2@prestashop.com"
+    And I add new address to customer "testCustomerShop2" with following details:
+      | Address alias    | test-customer-address              |
+      | First name       | testFirstName                      |
+      | Last name        | testLastName                       |
+      | Address          | Work address st. 1234567890        |
+      | City             | Birmingham                         |
+      | Country          | United States                      |
+      | State            | Alabama                            |
+      | Postal code      | 12345                              |
+    Then customer "testCustomerShop2" has address in "US" country
+    # Create Order
+    # Context : Shop 2
+    When I create an empty cart "cart_product_A" for customer "testCustomerShop2"
+    And I select "US" address as delivery and invoice address for customer "testCustomerShop2" in cart "cart_product_A"
+    And I add 2 products "Product A" to the cart "cart_product_A"
+    Then the remaining available stock for product "Product A" should be 98
+    ## 100 - 2 (in cart)
```

Comment:
Actually as long as the cart is not ordered the stock shouldn't be updated, what you are getting back here is the "virtual stock" which uses the cart as an argument I'm not sure it's the appropriate way to test the stock here especially since you are testing order features So you shouldn't rely on the "current cart" which can be modified as soon as you create a new cart (it's adapted, or at least convenient, when testing the cart but here I don't think it's ideal)

It would be more accurate to test the real stock which is independent from the cart you'd better use this step instead:
`Then the available stock for product "Test Duplicate Product" should be 100` (from `ProductFeatureContext::actualQuantityOfProductNamedShouldBe`)

Reply:
Fixed